chore(flags): Add `last_called_at` field to `FeatureFlag` model

haacked opened the pull request:
## Problem

There are many types of stale flags. One is a flag that isn't being called by code. So we want to track when the last time a flag was called.


<!-- Who are we building for, what are their needs, why is this important? -->

<!-- Does this fix an issue? Uncomment the line below with the issue ID to automatically close it when merged -->
<!-- Closes #ISSUE_ID -->

## Changes

Add `last_called_at DateTimeField` to track when a feature flag was last called. This enables identification of potentially stale feature flags that may represent dead code paths.

~Adds a Temporal workflow to update that field every 30 minutes by looking at up to 100,000 of the most recent `$feature_flag_called` events during an iteration.~

Going to add a simple Celery task as a follow-up PR.

## How did you test this code?

Manually
Unit Tests
<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

dmarticus commented on posthog/temporal/feature_flag_sync/workflow.py:
```diff
@@ -0,0 +1,219 @@
+        )
+
+        # Query ClickHouse for flag usage since last sync
+        # Limit 100000 for insurance against large datasets and memory issues during a surge
```
do we think it's worth batching transactions here just in case? might help us even more with insurance.

Not blocking, just figured I'd bring it up.

dmarticus commented on posthog/temporal/feature_flag_sync/workflow.py:
```diff
@@ -0,0 +1,219 @@
+        return result
+
+
+def get_last_sync_timestamp(redis_client, lookback_days=1) -> datetime:
```
can we make the `lookback_days` configurable with an environment variable + default

dmarticus commented on posthog/temporal/schedule.py:
```diff
@@ -127,6 +128,30 @@ async def create_upgrade_queries_schedule(client: Client):
+            id="feature-flag-sync-schedule",
+            task_queue=GENERAL_PURPOSE_TASK_QUEUE,
+        ),
+        spec=ScheduleSpec(intervals=[ScheduleIntervalSpec(every=timedelta(minutes=30))]),
```
If activity takes >30min, you'll have concurrent runs. Add:

```
policy=SchedulePolicy(
    overlap=ScheduleOverlapPolicy.BUFFER_ONE,  # or SKIP
)
```

to the schedule

dmarticus reviewed (approved):
I'm stamping because I trust you, but I called out some specific things that I think you should either (a) fix or (b) at least consider again.

In addition to those, I noticed the following magic numbers

```
batch_size=1000
LIMIT 100000
timedelta(minutes=10)
```

consider extracting them to constants

```
BULK_UPDATE_BATCH_SIZE = 1000
CLICKHOUSE_MAX_RESULTS = 100_000
ACTIVITY_TIMEOUT_MINUTES = 10
```

or making them environment variables if that makes more sense.

haacked committed:
Add migration for `featureflag.last_called_at` column

haacked committed:
Add last_called_at field to FeatureFlag model

Add `last_called_at` (`DateTimeField`) to `posthog_featureflag` (and the `FeatureFlagSerializer`) to track when a feature flag was last called. This enables identification of potentially stale feature flags that may represent dead code paths.

haacked committed:
Update snapshots and tests
